workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncherWindow.cs

```diff
@@ -522,82 +743,164 @@ private void DrawSpinner(float value, Rect rect)
 
         private class TaskManager
         {
-            public bool IsUploading { get; private set; }
-            public bool IsLaunching { get; private set; }
+            public bool IsActive => isUploading || isLaunching || isListing || isStopping || isAuthing;
 
             public readonly List<WrappedTask<RedirectedProcessResult, AssemblyConfig>> CompletedUploadTasks = new List<WrappedTask<RedirectedProcessResult, AssemblyConfig>>();
-            public readonly List<WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, BaseDeploymentConfig>> CompletedLaunchTasks = new List<WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, BaseDeploymentConfig>>();
+            public readonly List<WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, (string, string, BaseDeploymentConfig)>> CompletedLaunchTasks = new List<WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, (string, string, BaseDeploymentConfig)>>();
+            public readonly List<WrappedTask<Result<List<DeploymentInfo>, Ipc.Error>, string>> CompletedListTasks = new List<WrappedTask<Result<List<DeploymentInfo>, Ipc.Error>, string>>();
+            public readonly List<WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, DeploymentInfo>> CompletedStopTasks = new List<WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, DeploymentInfo>>();
+            public readonly List<WrappedTask<RedirectedProcessResult, int>> CompletedAuthTasks = new List<WrappedTask<RedirectedProcessResult, int>>();
 
             private WrappedTask<RedirectedProcessResult, AssemblyConfig> uploadTask;
-            private WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, BaseDeploymentConfig> launchTask;
+            private WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, (string, string, BaseDeploymentConfig)> launchTask;
             private WrappedTask<Result<RedirectedProcessResult, Ipc.Error>, DeploymentInfo> stopTask;
             private WrappedTask<Result<List<DeploymentInfo>, Ipc.Error>, string> listTask;
+            private WrappedTask<RedirectedProcessResult, int> authTask;
 
-            private Queue<(string, string, BaseDeploymentConfig)> queuedLaunches = new Queue<(string, string, BaseDeploymentConfig)>();
+            private bool isUploading;
```

Comment:
Aren't all of these bools redundant state that could be inferred from either the specific task being non-`null`, or a queue being non-empty?

Reply:
Yes - this needs another pass and can be simplified a lot imo